broker-daemon/models/block-order.js

```diff
@@ -160,7 +160,24 @@ class BlockOrder {
    */
   fail () {
     this.status = BlockOrder.STATUSES.FAILED
+    return this
+  }
+
+  /**
+   * Move the block order to a completed status
+   * @return {BlockOrder} Modified block order instance
+   */
+  complete () {
+    const activeOrders = this.openOrders.filter(o => o.status !== BlockOrder.STATUSES.COMPLETED)
```

Comment:
Can't a block order be completed even if an underlying order was cancelled?

Reply:
Yup. im missing this case